astor/string_repr.py

```diff
@@ -37,19 +38,13 @@ def _get_line(current_output):
         find the start of the current line,
         and return the entire line.
     """
-    myline = []
-    index = len(current_output)
-    while index:
-        index -= 1
-        try:
-            s = str(current_output[index])
-        except:
-            raise
-        myline.append(s)
-        if '\n' in s:
+    for index in range(len(current_output) - 1, -1, -1):
+        if '\n' in current_output[index]:
             break
-    myline = ''.join(reversed(myline))
-    return myline.rsplit('\n', 1)[-1]
+
+    myline = current_output[index:]
+    myline[0] = myline[0].rsplit('\n', 1)[-1]
+    return ''.join(myline)
 
```

Comment:
The optimization in this function also was a huge performance win.  Not calling `str()` was very helpful, and the loop optimization, and optimization to only split one of the substrings rather than the entire string were all useful.